Assets/_BForBoss/_Core/Scripts/Managers/ContextManager.cs

```diff
@@ -0,0 +1,11 @@
+using UnityEngine;
+
+namespace BForBoss
+{
+    /// <summary>
+    /// Base manager for handling and managing all context between Bosses, their labors and the environment
+    /// </summary>
+    public abstract class ContextManager : MonoBehaviour
```

Comment:
imo, don't need a parent abstract class just yet. Unless you already have a clear vision of what the other boss behaviours are like and have a plan for this class.  

i rather have 3+ of the same behaviours and then abstracting out, than trying to predict its usage - leading to more tech debt if it's not used.

Reply:
This is more of an opinion. It's up to you.

Reply:
Good point, Initially I had an idea of creating some parental logic but couldn't justify it in the end, will remove it!